src/frontend/src/pages/FinancePage/ReimbursementRequestDetailPage/SubmitToSaboModal.tsx

```diff
@@ -65,55 +81,106 @@ const SubmitToSaboModal = ({ open, setOpen, reimbursementRequest }: SubmitToSabo
-          <DetailDisplay label="Zip Code" content={userInfo.zipcode} />
+          <Box display="flex" alignItems="center">
+            <DetailDisplay label={'Zip Code'} content={userInfo.zipcode}></DetailDisplay>
+            <CopyToClipboardButton msg={userInfo.zipcode} />
+          </Box>
         </Grid>
       </Grid>
       <Grid container spacing={1} sx={{ marginTop: 2 }}>
-        <Grid item xs={5}>
-          <DetailDisplay label="Date Of Expense" content={datePipe(dateOfExpense)} />
+        <Grid item xs={6}>
+          <Box display="flex" alignItems="center">
+            <DetailDisplay label={'Date Of Expense'} content={datePipe(dateOfExpense)}></DetailDisplay>
+            <CopyToClipboardButton msg={datePipe(dateOfExpense)} />
+          </Box>
         </Grid>
         <Grid item xs={7}>
-          <DetailDisplay label="Total Expense" content={`$${centsToDollar(totalCost)}`} />
+          <Box display="flex" alignItems="center">
+            <DetailDisplay label={'Total Expense'} content={`$${centsToDollar(totalCost)}`}></DetailDisplay>
+            <CopyToClipboardButton msg={`$${centsToDollar(totalCost)}`} />
+          </Box>
         </Grid>
         <Grid item xs={12}>
-          <DetailDisplay label="Expense Description" content={`${vendor.name}[${centsToDollar(totalCost)}]`} />
+          <Box display="flex" alignItems="center">
+            <DetailDisplay
+              label={'Expense Decription'}
+              content={`${vendor.name}[${centsToDollar(totalCost)}]`}
+            ></DetailDisplay>
+            <CopyToClipboardButton msg={`${vendor.name}[${centsToDollar(totalCost)}]`} />
+          </Box>
         </Grid>
       </Grid>
       <Grid container spacing={1} sx={{ marginTop: 2 }}>
         <Grid item xs={12}>
-          <DetailDisplay label="Business Purpose" content={filteredProductsNames} />
+          <Box display="flex" alignItems="center">
+            <DetailDisplay label={'Business Purpose'} content={filteredProductsNames}></DetailDisplay>
+            <CopyToClipboardButton msg={filteredProductsNames} />
+          </Box>
         </Grid>
-        <Grid item xs={6}>
-          <DetailDisplay label="SABO Form Index" content={codeAndRefundSourceName(reimbursementRequest.account)} />
+        <Grid item xs={7}>
+          <Box display="flex" alignItems="center">
+            <DetailDisplay
+              label={'SABO Form Index'}
+              content={codeAndRefundSourceName(reimbursementRequest.account)}
+            ></DetailDisplay>
+            <CopyToClipboardButton msg={codeAndRefundSourceName(reimbursementRequest.account)} />
+          </Box>
         </Grid>
         <Grid item xs={6}>
-          <DetailDisplay label="Expense Type" content={`${expenseType.code} - ${expenseType.name}`} />
+          <Box style={{ display: 'flex', alignItems: 'center' }}>
+            <DetailDisplay label={'Expense Type'} content={`${expenseType.code} - ${expenseType.name}`}></DetailDisplay>
```

Comment:
instead of doing this in all these places, you should add a prop to detail display called `copyButton` that adds a copy button next to it like you're doing here